web/packages/teleterm/src/ui/ConnectMyComputer/DocumentConnectMyComputer/Setup.story.tsx

```diff
@@ -56,7 +57,7 @@ export function Success() {
     makeServer();
   // Report the agent as running so that the autostart behavior doesn't kick in and attempt to start
   // the agent over and over.
-  appContext.mainProcessClient.subscribeToAgentUpdate = (
+  (appContext.mainProcessClient as MainProcessClient).subscribeToAgentUpdate = (
```

Comment:
This is somewhat silly. `appContext.mainProcessClient` is now of `MockMainProcessClient` type. Its `subscribeToAgentUpdate` methods, unlike that method in its superclass, takes zero arguments not two, so TS wouldn't allow me to override it like this.

---

As a side note, I don't understand why TS isn't able to infer method arguments from a superclass. For example, I had to manually type of arg types here despite the method existing on the original type.

https://github.com/gravitational/teleport/blob/46804c2cc8f26e2bb6a18c8acafea577c2715522/web/packages/teleterm/src/mainProcess/fixtures/mocks.ts#L66-L71

Reply:
Sure, that's a good idea. In an ideal world, I'd rather declare that `mainProcessClient` on `MockAppContext` is `MockMainProcessClient`, but having to write that extra assertions is not worth it.